### PR TITLE
use correct way to do the update to new major versions

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/issues/3908

This is best reviewed like this: https://github.com/nextcloud/all-in-one/pull/3915/files?w=1

### CODE REVIEW

File: Containers/nextcloud/entrypoint.sh [hunk 1]
```diff
@@ -258,26 +258,21 @@ DATADIR_PERMISSION_CONF
             if [ "$INSTALL_LATEST_MAJOR" = yes ]; then
                 php /var/www/html/occ config:system:set updater.release.channel --value=beta
```
Comment (blizzz) [inline Containers/nextcloud/entrypoint.sh:259]: still want to stay on beta?

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh:259]: Ah right, forgot to remove that

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh]: done with https://github.com/nextcloud/all-in-one/pull/3915/commits/6412aa3e0f4262e5fa6b163be7efb3bbd2ed5e2e

File: Containers/nextcloud/entrypoint.sh [hunk 2]
```diff
@@ -256,31 +256,24 @@ DATADIR_PERMISSION_CONF
...
             if [ "$INSTALL_LATEST_MAJOR" = yes ]; then
-                php /var/www/html/occ config:system:set updater.release.channel --value=beta
                 php /var/www/html/occ config:system:set updatedirectory --value="/nc-updater"
-                php /var/www/html/updater/updater.phar --no-interaction
+                INSTALL_DAT="$(php /var/www/html/occ config:app:get core installedat)"
+                if [ -n "$INSTALL_DAT" ]; then
+                    # Set the installdat to 00 which will allow to skip staging and install the next major directly
+                    # shellcheck disable=SC2001
+                    INSTALL_DAT="$(echo "$INSTALL_DAT" | sed "s|[0-9][0-9]$|00|")"
+                    php /var/www/html/occ config:app:set core installedat --value="$INSTALL_DAT"
```
Comment (blizzz) [inline Containers/nextcloud/entrypoint.sh:265]:
```suggestion
                INSTALLED_AT="$(php /var/www/html/occ config:app:get core installedat)"
                if [ -n "${INSTALLED_AT}" ]; then
                    # Set the installdat to 00 which will allow to skip staging and install the next major directly
                    # shellcheck disable=SC2001
                    INSTALLED_AT="$(echo "${INSTALLED_AT}" | sed "s|[0-9][0-9]$|00|")"
                    php /var/www/html/occ config:app:set core installedat --value="${INSTALLED_AT}" 
```

**Review by blizzz — APPROVED** [review]
with one suggestion

**Comment by szaimen** [comment]
Actually, I just figured that we need the version check in some cases...

**Comment by szaimen** [comment]
> Actually, I just figured that we need the version check in some cases...

done

**Review by blizzz — APPROVED** [review]

**Comment by szaimen** [comment]
Thanks Arthur! :)

**Pull request merged**